src/Octoshift/AdoApi.cs

```diff
@@ -284,139 +286,127 @@ public virtual async Task ShareServiceConnection(string adoOrg, string adoTeamPr
         {
             var url = $"https://dev.azure.com/{adoOrg}/_apis/serviceendpoint/endpoints/{serviceConnectionId}?api-version=6.0-preview.4";
 
-            var payload = @"
-[{
-    ""name"": ""ADO_ORGANIZATION-ADO_TEAM_PROJECT"",
-	""projectReference"": {
-        ""id"": ""ADO_PROJECT_ID"",
-		""name"": ""ADO_TEAM_PROJECT""
-    }
-}]";
-
-            payload = payload.Replace("ADO_ORGANIZATION", adoOrg);
-            payload = payload.Replace("ADO_TEAM_PROJECT", adoTeamProject);
-            payload = payload.Replace("ADO_PROJECT_ID", adoTeamProjectId);
+            var payload = new[]
+            {
+                new
+                {
+                    name = $"{adoOrg}-{adoTeamProject}",
+                    projectReference = new
+                    {
+                        id = adoTeamProjectId,
+                        name = adoTeamProject
+                    }
+                }
+            };
 
             await _client.PatchAsync(url, payload);
         }
 
-        public virtual async Task<AdoPipeline> GetPipeline(string org, string teamProject, int pipelineId)
+        public virtual async Task<(string defaultBranch, string clean, string checkoutSubmodules)> GetPipeline(string org, string teamProject, int pipelineId)
```

Comment:
Since behind the scenes a value tuple is actually a `struct` all properties should be pascal cased. Specially when we expose a value tuple. The only exception is when we use the properties in a method where they are actually local variables.

```suggestion
        public virtual async Task<(string DefaultBranch, string Clean, string CheckoutSubmodules)> GetPipeline(string org, string teamProject, int pipelineId)
```